src/Compilers/CSharp/Portable/Lowering/Instrumentation/DynamicAnalysisInjector.cs

```diff
@@ -404,7 +400,7 @@ private static bool ReturnsValueWithinExpressionBodiedConstruct(BoundReturnState
                 returnStatement.ExpressionOpt != null &&
                 returnStatement.ExpressionOpt.Syntax != null)
             {
-                SyntaxKind parentKind = returnStatement.ExpressionOpt.Syntax.Parent.Kind();
+                SyntaxKind parentKind = returnStatement.ExpressionOpt.Syntax.Parent!.Kind();
```

Comment:
can you explain why we know the Parent is never null?

Reply:
The parent of syntax node that comes out of parser (lowering only deals with such nodes) is never null unless it is CompilationUnit.